cypress/e2e/e2e.cy.js

```diff
@@ -0,0 +1,140 @@
+describe('Teste E2E EngageSphere', () => {
+    beforeEach(() => {
+        cy.visit('http://localhost:3000')
+        cy.contains('button', 'Accept').click();
+    })
+
+    it('Mantém os filtros ao voltar da visualização de detalhes do cliente', () => {
+        cy.get('select[data-testid="size-filter"]').select('Very Large Enterprise')
+        cy.get('button[aria-label="View company: Lowe Co"]').click()
+        cy.get('button').contains('Back').should('be.visible')
+        cy.contains('button', 'Back').click();
+        cy.contains('Below is our customer list.').should('be.visible')
+        cy.get('select[data-testid="size-filter"]').should('have.value', 'Very Large Enterprise');
+    })
+
+    it('Retorna à lista de clientes ao clicar no botão "Voltar"', () => {
+        cy.get('button[aria-label="View company: Lowe Co"]').click();
+        cy.get('button').contains('Back').should('be.visible')
+        cy.contains('button', 'Back').click();
+        cy.contains('Below is our customer list.').should('be.visible')
+
+    })
+
+
+    it('Exibie o rodapé com o texto e links corretos', () => {
+        cy.get('[data-testid="footer"]').should('be.visible')
+
+        cy.get('[data-testid="footer"] a').eq(0).should('be.visible')
+        cy.get('[data-testid="footer"] a').eq(0).should('have.attr', 'href', 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem');
+
+        cy.get('[data-testid="footer"] a').eq(1).should('be.visible')
+        cy.get('[data-testid="footer"] a').eq(1).should('have.attr', 'href', 'https://udemy.com/user/walmyr');
+
+        cy.get('[data-testid="footer"] a').eq(2).should('be.visible')
+        cy.get('[data-testid="footer"] a').eq(2).should('have.attr', 'href', 'https://talkingabouttesting.com');
+
+        cy.get('[data-testid="footer"] a').eq(3).should('be.visible')
+        cy.get('[data-testid="footer"] a').eq(3).should('have.attr', 'href', 'https://youtube.com/@talkingabouttesting');
+
+    })
+
+    it('Exibe a saudação "Hi, there" quando nenhum nome é fornecido', () => {
+        cy.get('h2').should('contain', 'Hi there!')
+    })
+
+    it('Exibe a saudação "Hi, Joe" quando o nome é fornecido', () => {
+        const name = 'Joe'
+        cy.get('#name').type(name)
+        cy.get('h2').should('contain', 'Hi ' + name + '!')
+    })
+
+    it('Exibe o cabeçalho com um título, alternador de tema e um campo de entrada de texto', () => {
+        cy.get('h1').should('be.visible')
+        cy.get('h1').should('contain', 'EngageSphere')
```

Comment:
Em vez desta e da linha acima, que tal algo como o seguinte:

```suggestion
        cy.contains('h1', 'EngageSphere').should('be.visible')
```